### PR TITLE
provider/aws: fix panic with SNS topic policy if omitted

### PR DESCRIPTION
**Pull request by catsby**
Builds of of #3751, thanks @bobtfish ! 

- removes `ForceNew: false`
- return `""` if `v` is nil in StateFunc

### CODE REVIEW

**Comment by stack72** [comment]
Looks good @catsby - merging this will fix my acceptance tests :)

File: builtin/providers/aws/resource_aws_sns_topic.go [hunk 1]
```diff
@@ -44,10 +44,12 @@ func resourceAwsSnsTopic() *schema.Resource {
...
 				Type:     schema.TypeString,
 				Optional: true,
-				ForceNew: false,
 				Computed: true,
 				StateFunc: func(v interface{}) string {
-					jsonb := []byte(v.(string))
+					if v == nil {
+						return ""
+					}
+					jsonb := []byte{}
```
Comment (rjeczalik) [inline builtin/providers/aws/resource_aws_sns_topic.go]: The value of `v` is ignored here. What about:

```
s, ok := v.(string)
if !ok || s == "" {
  return ""
}
jsonb := []byte(s)
```

Comment (catsby) [inline builtin/providers/aws/resource_aws_sns_topic.go]: :scream: great catch, thanks @rjeczalik, updated in 15533dc

**Comment by phinze** [comment]
LGTM

**Pull request merged**